### PR TITLE
Migrate DQMEDAnalyzers from `stream` to `one` modules

### PR DESCRIPTION
**Pull request by fwyzard**
Migrate all DQMEDAnalyzer modules from `edm::stream::EDAnalyzer<...>` to `edm::one::EDAnalyzer<...>`.

This should significantly reduce the memory usage of a multithreaded DQM job, at the price of preventing the framework from analysing multiple runs or lumisections at the same time.
    
Lumisection-based histograms are handled in the `postModuleGlobalEndLumi` signal of the `DQMStore`.

If `DQMEDanalyzer` modules are migrated from `EDAnalyzer` to `EDProduce<Accumulator>`, that signal should be removed and the handling moved to the modules' `endLuminosityBlockProduce` method.

### CODE REVIEW

**Comment by fwyzard** [comment]
First attempt at a simpler, partial migration for #22157 .

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by schneiml** [comment]
relative to #22157, I see
- pro: no mess with config changes.
- pro: no issues related to the new `edm::Accumulator` feature.
- con: these two were not the problems discussed in #22157.
- con: no idea if there is a disadavantage in using `EDAnalyzer`, what does @Dr15Jones think?

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by schneiml** [comment]
This solution looks good. Since now fallback to stream modules will be needed, all the related code in `DQMEDAnalyzer`/`DQMStore` can be removed, as done here. There is still an issue with per-lumi histograms which can be fixed by adapting the output module.

I will assemble a new PR based on this commit, and split out the config changes of #22157 into #22243 . I will assemble another PR based on this commit and some more from #22157, plus the output module change, to migrate to `edm::one::EDAnalyzer`. These two are then orthogonal. Once both are in, it is to switch to `edm::one::EDProducer<Accumulator>`, if this is useful (@Dr15Jones ?). This can happen immediatly, or once we have the proper token-based dependencies implemented.

I suggest to close this PR.

**Comment by fwyzard** [comment]
Mhm.

Changing the output modules will make it impossible to later support concurrent lumisections, also in global modules.

If I can think of a solution that preserves that possibility, I will push it here.

**Comment by schneiml** [comment]
It is not like such changes can't be rolled back. At the moment, the goal is to keep the changes minimal and save for data taking, more changes will very likely follow over summer.

But yes, I you can come up with a solution, we are happy to use it!

**Comment by fwyzard** [comment]
@schneiml , this should have a higher chance of working, if you want to have a look

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by schneiml** [comment]
@fwyzard do I see correctly that this PR is not safe to use atm (even though the tests pass), since there is no `edm::Accumulator` migration yet?

We will do the renaming now (since it should happen at some point anyways), and this PR can probably be closed, to avoid confusion.

**Comment by fwyzard** [comment]
There is no migration from `Analyzer` to `Producer` here, hence possibility to use the `edm::Accumulator` feature.

**Comment by fwyzard** [comment]
@schneiml @dmitrijus please have a look

What workflow can I use to validate the changes ?

**Comment by fwyzard** [comment]
@cmsbuild please test

**Comment by dmitrijus** [comment]
+1

**Comment by fabiocos** [comment]
+1

File: DQMServices/Core/src/DQMStore.cc [hunk 1]
```diff
@@ -856,7 +707,7 @@ DQMStore::book_(const std::string &dir, const std::string &name,
   {
     // Create and initialise core object.
     assert(dirs_.count(dir));
-    MonitorElement proto(&*dirs_.find(dir), name, run_, streamId_, moduleId_);
+    MonitorElement proto(&*dirs_.find(dir), name, run_, moduleId_);
```
Comment (schneiml) [inline DQMServices/Core/src/DQMStore.cc]: @fwyzard ugh... What have you done there? Did you also remove the `streamId` from the `MonitorElement` constructor? It should be removed, but the change is not in this PR, so now `streamId`s and `moduleId`s are messed up!

**Comment by fwyzard** [comment]
I thought I did fix all the constructors as well... I can have a look
tomorrow.

**Comment by fwyzard** [comment]
@schneiml from a cursory look, I think I have updated all the calls to the `MonitorElement` constructors.

If you can point me where you see an inconsistency between the `streamId` and `moduleId` I will try to fix it.

**Comment by schneiml** [comment]
@fwyzard streamId is still in the constructor declaration, so you are passing moduleId *instead* of streamId. Which would be sort of fine if it was used consistently, but some places do nontrival queries on the moduleId (e.g., `save()`). Just remove streamId from the constructor and see where it fails to compile...

**Comment by fwyzard** [comment]
Ah, I see... and it was not spot before by the compiler because of all the default parameters.
OK, let's see if #22364 helps.

**Pull request merged**